src/dipdup/datasources/factory.py

```diff
@@ -0,0 +1,105 @@
+import warnings
+
+from dipdup.config import BcdDatasourceConfig
+from dipdup.config import CoinbaseDatasourceConfig
+from dipdup.config import DipDupConfig
+from dipdup.config import HttpDatasourceConfig
+from dipdup.config import IpfsDatasourceConfig
+from dipdup.config import MetadataDatasourceConfig
+from dipdup.config import TzktDatasourceConfig
+from dipdup.datasources.bcd.datasource import BcdDatasource
+from dipdup.datasources.coinbase.datasource import CoinbaseDatasource
+from dipdup.datasources.const import DatasourceKind
+from dipdup.datasources.datasource import Datasource
+from dipdup.datasources.datasource import HttpDatasource
+from dipdup.datasources.ipfs.datasource import IpfsDatasource
+from dipdup.datasources.metadata.datasource import MetadataDatasource
+from dipdup.datasources.tzkt.datasource import TzktDatasource
+
+
+class DatasourceFactory:
```

Comment:
I don't like the idea to add a 105 LOC module for a single factory. Something like this should be more than enough:

```python
@classmethod
def build(cls, name: str, config: DatasourceConfig) -> Datasource:
    datasource = cls._build_datasource(config)
    datasource.set_logger(name)
    datasource.set_user_agent(config.package)

@classmethod
def _build_datasource(config: DatasourceConfig) -> Datasource:
    if isinstance(config, TzktDatasourceConfig):
        return TzktDatasource(
            url=datasource_config.url,
            http_config=datasource_config.http,
            merge_subscriptions=config.advanced.merge_subscriptions,
        )
    if isinstance(config, ...):
        ...
    raise NotImplementedError
```